diff --git a/Configurations/15-android.conf b/Configurations/15-android.conf
index 1de6b7a..285e87c 100644
--- a/Configurations/15-android.conf
+++ b/Configurations/15-android.conf
@@ -223,13 +223,13 @@ my %targets = (
         # Newer NDK versions reportedly require additional -latomic.
         #
         inherit_from     => [ "android" ],
-        bn_ops           => add("RC4_CHAR"),
+        bn_ops           => "BN_LLONG RC4_CHAR",
         asm_arch         => 'armv4',
         perlasm_scheme   => "void",
     },
     "android-arm64" => {
         inherit_from     => [ "android" ],
-        bn_ops           => add("RC4_CHAR"),
+        bn_ops           => "SIXTY_FOUR_BIT_LONG RC4_CHAR",
         asm_arch         => 'aarch64',
         perlasm_scheme   => "linux64",
     },
@@ -259,14 +259,14 @@ my %targets = (
     "android-x86" => {
         inherit_from     => [ "android" ],
         CFLAGS           => add(picker(release => "-fomit-frame-pointer")),
-        bn_ops           => add("RC4_INT"),
+        bn_ops           => "BN_LLONG RC4_INT",
         asm_arch         => 'x86',
         perlasm_scheme   => "android",
         ex_libs          => add(threads("-latomic")),
     },
     "android-x86_64" => {
         inherit_from     => [ "android" ],
-        bn_ops           => add("RC4_INT"),
+        bn_ops           => "SIXTY_FOUR_BIT_LONG RC4_INT",
         asm_arch         => 'x86_64',
         perlasm_scheme   => "elf",
     },
